examples/sampling/pyg/test.py

```diff
@@ -0,0 +1,167 @@
+"""
+This script demonstrates the training and testing of a GraphSAGE model for node classification on large graphs, leveraging the 
+GraphBolt framework for efficient data handling and mini-batch processing.
+
+Key Features:
+- Implements the GraphSAGE model, a scalable GNN, for node classification on large graphs.
+- Utilizes GraphBolt, an efficient framework for large-scale graph data processing.
+- Integrates with PyTorch Geometric for building and training the GraphSAGE model.
+- The script is well-documented, providing clear explanations at each step.
+
+This flowchart describes the main functional sequence of the provided example.
+main: 
+
+main
+│
+├───> Load and preprocess dataset (GraphBolt)
+│     │
+│     └───> Utilize GraphBolt's BuiltinDataset for dataset handling
+│
+├───> Instantiate the SAGE model (PyTorch Geometric)
+│     │
+│     └───> Define the GraphSAGE model architecture
+│
+├───> Train the model
+│     │
+│     ├───> Mini-Batch Processing with GraphBolt
+│     │     │
+│     │     └───> Efficient handling of mini-batches using GraphBolt's utilities
+│     │
+│     └───> Training Loop
+│           │
+│           ├───> Forward and backward passes
+│           │
+│           └───> Parameters optimization
+│
+└───> Evaluate the model
+      │
+      └───> Performance assessment on validation and test datasets
+            │
+            └───> Accuracy and other relevant metrics calculation
+
+
+"""
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from dgl.graphbolt import (
+    BuiltinDataset,
+    CopyTo,
+    DataLoader,
+    FeatureFetcher,
+    ItemSampler,
+)
+from torch_geometric.nn import GCNConv
+
+
+dataset = BuiltinDataset("ogbn-arxiv").load()
+graph = dataset.graph
+feature = dataset.feature
+train_set = dataset.tasks[0].train_set
+valid_set = dataset.tasks[0].validation_set
+test_set = dataset.tasks[0].test_set
+num_classes = dataset.tasks[0].metadata["num_classes"]
+
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+
+def create_sampler_and_loader(dataset_set):
+    sampler = ItemSampler(dataset_set, batch_size=1024, shuffle=True)
+    datapipe = sampler.sample_neighbor(graph, [4, 4])
+    datapipe = FeatureFetcher(datapipe, feature, node_feature_keys=["feat"])
```

Comment:
use the function-like style which is the recommended way.

Reply:
cool